File: 0274-h-index/0274-h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        std::sort(citations.begin(), citations.end());
        int h = 0;
        int n = citations.size();
        for (int i = 0; i < n; i++) {
            int candidate = min(citations[i], n - i);
            if (h < candidate) {
                h = candidate;
            }
        }
        return h;
    }
};

File: top-interview-150/0274-h-index/0274-h-index.py
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        citations.sort()
        
        h = 0
        n = len(citations)
        for i in range(n):
            candidate = min(citations[i], n - i)
            if h < candidate:
                h = candidate
                
        return h